src/plugins/kibana_react/public/validated_range/validated_dual_range.tsx

```diff
@@ -47,7 +66,11 @@ export class ValidatedDualRange extends Component {
     return null;
   }
 
-  _onChange = value => {
+  state: State = {
+    value: ['0', '0'],
```

Comment:
Is setting a default state required by typescript? State should be configured by `getDerivedStateFromProps`. Would be nice to keep initial state as `{}` since its derived from props.

Reply:
Unfortunately yes in this case. `EuiDualRange` doesn't accept an `undefined` value. Its value refers back to `this.state.value`. For this reason, I also couldn't give `this.state.value` a type of `Value` and instead had to set to `[ValueMember, ValueMember]`.

Reply:
How about this then

```
// @ts-ignore state populated by getDerivedStateFromProps
state = {};
```

Providing a default value for `value` could result in bugs if the default value equals the prop.value and `getDerivedStateFromProps` does not init the other parts of state.

Reply:
We'd still be passing an initially undefined value through to `EuiDualRange` on line 119, which is really where we're getting the TypeScript error. Before in js, we were passing through `undefined` and basically "got away with it" because it was handled on the other side. Now it blows up the linter if we make `value` optional or allow undefined. Alternatively we could clear the initial state as you've written it there and move this comment down to line 119, but to me it seems less buggy to me to handle it on our side as is now knowing that any place this component is loaded and provides any sort of value it will be immediately overwritten

Reply:
I just tested locally and changing the below to the `state = {};` passes ts-linter. You can run the linter locally with the command `node scripts/type_check --project=x-pack/tsconfig.json`. `getDerivedStateFromProps` is called before `render` so `this.state.value` will exist and get passed to EuiDualRange.  

```
state: State = {
    value: ['0', '0'],
}
```

to 

```
// @ts-ignore state populated by getDerivedStateFromProps
state = {};
```

Reply:
https://reactjs.org/docs/react-component.html

<img width="400" alt="Screen Shot 2020-03-12 at 7 02 42 PM" src="https://user-images.githubusercontent.com/373691/76580048-16485d00-6494-11ea-81aa-617470d77a42.png">

Reply:
Okay, thanks for the testing and the explanation. I'm not sure why it was getting flagged by my IDE linter. I'll update it